packages/babel-types/src/definitions/core.ts

```diff
@@ -1233,7 +1233,12 @@ defineType("VariableDeclarator", {
       validate:
         !process.env.BABEL_8_BREAKING && !process.env.BABEL_TYPES_8_BREAKING
           ? assertNodeType("LVal")
-          : assertNodeType("Identifier", "ArrayPattern", "ObjectPattern"),
+          : assertNodeType(
```

Comment:
In the `VariableDeclaration` builder, can we disallow `VoidPattern` for `var`/`let`/`const`?

Reply:
https://github.com/babel/babel/pull/17163/commits/3bdcdd3f82ab2648f1ca76f79a3f7a3051f38358